crates/analyzer/src/traversal/expressions.rs

```diff
@@ -850,7 +833,15 @@ fn expr_call_type_constructor(
                 Location::Value,
             ))
         }
-        _ => Err(FatalError::new()),
+        Type::Struct(_) => unreachable!(), // handled above
+
+        // TODO: These type should be handled explicitly, either here or in expr_name_call_type.
```

Comment:
Mind opening an issue about this?

Reply:
I've addressed these TODOs in a separate branch; I'll open a PR today.